### ISSUE TITLE
Unterminated JSX Contents

### ISSUE THREAD

**Issue by stephancasas** [lukehaas/RunJS#306]
The following function definition is throwing an `Unterminated JSX contents` error:

```ts
const attrSelector = (attr: string): string =>
  `[${[...attr]
    .map((char) => `\\${char.charCodeAt(0).toString(16)} `)
    .join('')}]`;
```

The function, though a bit dense, is valid TS. I have the `preset-react` option disabled, but the error is still thrown. Using the playground on [typescriptlang.org](https://www.typescriptlang.org/play), I do have to specify JSX as `none`, but after that, it works perfectly.

**Comment by lukehaas** [comment]
I'm not seeing the issue you're describing. This function is not JSX so this error is more likely related to some other code.

This is what I'm seeing:
![Screenshot 2021-08-22 at 12 20 13](https://user-images.githubusercontent.com/741344/130353230-5e1e6fe7-d114-4148-b3ab-74a0e0b78371.png)

**Comment by stephancasas** [comment]
I tested again, and it looks like I didn't go down far enough in my process of elimination to determine the issue. The function declared above the one I gave was this:

```ts
const str2el = (str: string): HTMLElement => {
  const div = document.createElement('div');
  div.innerHTML = str;
  return <HTMLElement>div.firstChild;
};
```

Never mind that the type cast `<HTMLElement>` isn't exactly correct 😬, but it does seem to be what's causing RunJS to read as JSX. Not a big deal. I can just use the `as` syntax instead, and cast the type correctly while I'm at it, haha.
